reflect/cache.go

```diff
@@ -234,32 +235,64 @@ func dominantField(fields []Field) (Field, bool) {
 	return fields[0], true
 }
 
+type fieldCacheEntry struct {
```

Comment:
I tweaked this because StructFieldMap is doing constant repeated work in generating the field map which remains static for the lifetime of the application. That's really wasteful and results in heap allocations that can be saved for other things